src/rebar3_hex_publish.erl

```diff
@@ -339,12 +324,79 @@ maybe_prompt(_Args, Message) ->
             abort
     end.
 
-create_tarball(#{metadata := Meta, files := Files}) ->
-    case hex_tarball:create(Meta, Files) of
-        {ok, #{tarball := Tarball, inner_checksum := _Checksum}} ->
-            Tarball;
-        Error ->
-            ?RAISE(Error)
+hex_opts(Opts) ->
+    lists:filter(fun({replace, _}) -> true;
```

Comment:
Folded `is_hex_opt` into the main function since it was really just spreading the mess around.